### PR TITLE
fix(select): sanitize user input before searching options

### PR DESCRIPTION
**Pull request by psamim**
## PR Checklist
Please check that your PR fulfills the following requirements:
- [x] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [x] Tests for the changes have been added or this is not a bug fix / enhancement
- [x] Docs have been added, updated, or were not required

## PR Type
What kind of change does this PR introduce?
```
[x] Bugfix
[ ] Enhancement
[ ] Documentation content changes
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

Issue Number: 
https://github.com/angular/material/issues/11854

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by psamim** [comment]
@googlebot I signed it!

File: src/components/select/select.js [hunk 1]
```diff
@@ -729,8 +729,14 @@ function SelectMenuDirective($parse, $mdUtil, $mdConstant, $mdTheming) {
           return optNodes[i];
         }
       }
+
     };
 
+    self.sanitize = function(term) {
+      if (!term) return term;
+      return term.replace(/[\\^$*+?.()|{}[]]/g, '\\$&');
+    }
```
[the remark this answers is not in the document]
Comment (psamim) [reply src/components/select/select.js]: @Splaktar Thanks! I applied the changes as described.

**Comment by psamim** [comment]
@Splaktar I applied the requested changes. Thanks for the review.

**Comment by Splaktar** [comment]
✅ Rebased.

**Pull request merged**